### PR TITLE
Ranking Criteria/BN rule updates

### PR DESCRIPTION
**Pull request by pishifat**
an unreasonable amount of threads this time

https://osu.ppy.sh/community/forums/topics/847666: standardizing semi-common metadata phrases
~~https://osu.ppy.sh/community/forums/topics/832111: minor wording adjustment to match how users respond to guidelines~~ removed this one, see thread
https://osu.ppy.sh/community/forums/topics/868716: moved from general to osu! specific
https://osu.ppy.sh/community/forums/topics/860714: added rule about applause sfx filetype
https://osu.ppy.sh/community/forums/topics/877557: clarified how source can be used in unclear situations
https://osu.ppy.sh/community/forums/topics/837952 and https://osu.ppy.sh/community/forums/topics/853985: edited bg regulations
https://osu.ppy.sh/community/forums/topics/871492: "must" changed to "should"
https://osu.ppy.sh/community/forums/topics/872345: clarified multi-mode bn exception
https://osu.ppy.sh/community/forums/topics/744298: clear sliderpath clarification
https://osu.ppy.sh/community/forums/topics/857987: general guideline cleanup
https://osu.ppy.sh/community/forums/topics/857982: osu! guideline cleanup
https://osu.ppy.sh/community/forums/topics/888625: reverted part of veto rule
https://osu.ppy.sh/community/forums/topics/889929: handling whitespaces in tags
https://osu.ppy.sh/community/forums/topics/857306 ctb thing 1
https://osu.ppy.sh/community/forums/topics/844454 ctb thing 2
https://osu.ppy.sh/community/forums/topics/889616 difficulty naming

---

### CODE REVIEW

File: wiki/Ranking_Criteria/en.md [hunk 1]
```diff
@@ -139,12 +139,12 @@ Glossary
 
 ### Rules
 #### Technical
-- **Metadata must be consistent across all difficulties of a beatmap set.**
-- **Guest mappers, storyboarders, skinners and hitsounders must be added to the tags of a beatmap set.** This is to give credit where credit is due and help others identify the main contributors of any given beatmap set. Usernames containing whitespaces should be written with underscores instead.
+- **Metadata must be consistent across all difficulties of a beatmapset.**
```
Comment (cl8n) [inline wiki/Ranking_Criteria/en.md]: you changed to "beatmapset" here but added "beatmap set" in BN rules, change all of em to beatmapset (might want to check other usages throughout the wiki too)

File: wiki/Ranking_Criteria/en.md [hunk 2]
```diff
@@ -139,12 +139,12 @@ Glossary
...
+- **Guest mappers, storyboarders, skinners and hitsounders must be added to the tags of a beatmapset.** This is to give credit where credit is due and help others identify the main contributors of any given beatmapset. Usernames containing single characters separated by whitespaces should replace whitespaces with underscores.
```
Comment (cl8n) [inline wiki/Ranking_Criteria/en.md]:
```suggestion
- **Guest mappers, storyboarders, skinners and hitsounders must be added to the tags of a beatmapset.** This is to give credit where credit is due and help others identify the main contributors of any given beatmapset. Usernames containing single characters separated by whitespaces must have the whitespaces replaced with underscores.
```
usernames don't make decisions, people do

File: wiki/Ranking_Criteria/osu!/en.md [hunk 3]
```diff
@@ -57,27 +54,21 @@ Overall rules and guidelines apply to every kind of osu! difficulty. Rhythm rela
...
--   **Avoid using similar combo colours in successive combos.** Players should be able to differentiate between different combos in a beatmap.
 -   **Spinner ends, slider ends, and slider reverses should have hit sound feedback.** If these are used to represent a held sound and do not align with a distinct sound, having no feedback is acceptable.
 -   **Avoid using sound samples for sliderslide, sliderwhistle, and spinnerspin which do not naturally loop.** These hit sounds are continuous, meaning that their files play from start to end and loop as one continuous sound for the length of the object, so using sound files with a clear impact for them might lead to unwanted side effects.
 -   **Avoid replacing the hit finish in soft/normal samplesets with frequently used custom hit sound samples.** Using these finishes to represent snare/bass drums or a song's melody can sound obnoxious for anyone disabling beatmap hit sounds. Replacing hit whistles/claps is recommended because those samples are used more often.
+-   **Avoid unjustified spikes in difficulty.** Difficulty should be representative of the song's intensity.
```
Comment (cl8n) [inline wiki/Ranking_Criteria/osu!/en.md]: seems like something for general RC?

File: wiki/Ranking_Criteria/osu!/en.md [hunk 4]
```diff
@@ -112,10 +103,16 @@ Difficulty-specific rules and guidelines do only apply to the difficulty level t
...
 -  **Use straightforward and easily understandable slider shapes.** New players may not understand how to follow more complex shapes.
 -  **Spinner recovery time should be at least 4 beats.** This is to ensure adequate time to click a hit object following a spinner.
 -  **Avoid spinners less than 4 beats.** Players need time to recognize that they have to begin spinning.
+-  **Avoid fully covering slider reverse arrows with hitobjects within 4 beats of the reverse arrow being reached.** This gives players enough time to recognize the reverse arrow.
+-  **Avoid perfectly overlapping slider bodies within 4 beats of each other.** Doing so can cause sliders to be misread as a circle due to obscuring the slider body.
+-  **Ensure that your combos are not unreasonably short or long.** Combos should reflect patterns expressed in the song, such as bars of music or vocal/instrumental phrases.
+-  **Avoid overlapping hit objects with other elements of the default and beatmap-specific skins.** This refers to all elements that are part of the interface and can be skinned.
+
+
```
Comment (cl8n) [inline wiki/Ranking_Criteria/osu!/en.md]: remove whitespace to save trees

File: wiki/Ranking_Criteria/osu!/en.md [hunk 5]
```diff
@@ -112,10 +103,16 @@ Difficulty-specific rules and guidelines do only apply to the difficulty level t
...
 -  **Avoid slider-only sections.** Aiming and following a lot of sliders in a row can be tiring for new players. In such cases, circles and rest moments without hit objects to click or follow should be used.
--  **Manipulating slider velocity is discouraged.** Slider velocity should only be changed for sections of music with different pacings and should not be drastically variable.
+-  **Frequently manipulating slider velocity is discouraged.** Slider velocity should only be changed for sections of music with different pacings and should not be drastically variable.
 -  **Use straightforward and easily understandable slider shapes.** New players may not understand how to follow more complex shapes.
 -  **Spinner recovery time should be at least 4 beats.** This is to ensure adequate time to click a hit object following a spinner.
 -  **Avoid spinners less than 4 beats.** Players need time to recognize that they have to begin spinning.
+-  **Avoid fully covering slider reverse arrows with hitobjects within 4 beats of the reverse arrow being reached.** This gives players enough time to recognize the reverse arrow.
```
Comment (cl8n) [inline wiki/Ranking_Criteria/osu!/en.md]:
```suggestion
-  **Avoid fully covering slider reverse arrows with hit objects within 4 beats of the reverse arrow being reached.** This gives players enough time to recognize the reverse arrow.
```

File: wiki/Ranking_Criteria/osu!/en.md [hunk 6]
```diff
@@ -171,6 +172,8 @@ If you use a Normal difficulty as the _lowest difficulty_ of your beatmapset, it
 -  **Avoid mixing 1/2 and 1/4 rhythms in the same stack.** Different rhythms of this density without spacing indication can cause reading issues.
 -  **Slider tick hit sounds are discouraged.** If you want to use them, then make sure that their volume is balanced (i.e. notably quieter than regular hit sounds). A very loud slider tick, especially when only used once or twice, can be extremely jarring.
 -  **Avoid spinners less than 2 beats.** Players need time to recognize that they have to begin spinning.
+-  **Avoid fully covering slider reverse arrows with hitobjects within 2 beats of the reverse arrow being reached.** This gives players enough time to recognize the reverse arrow.
```
Comment (cl8n) [inline wiki/Ranking_Criteria/osu!/en.md]:
```suggestion
-  **Avoid fully covering slider reverse arrows with hit objects within 2 beats of the reverse arrow being reached.** This gives players enough time to recognize the reverse arrow.
```

File: wiki/Ranking_Criteria/osu!/en.md [hunk 7]
```diff
@@ -135,14 +132,18 @@ Difficulty-specific rules and guidelines do only apply to the difficulty level t
...
 -  **Stacks are acceptable, but avoid switching between 1/1 and 1/2 stacks repeatedly.** This may cause reading issues for new players, since two different rhythms are visually the same.
--  **Manipulating slider velocity is discouraged.** Slider velocity should only be changed for sections of music with different pacings and should not be drastically variable.
+-  **Frequently manipulating slider velocity is discouraged.** Slider velocity should only be changed for sections of music with different pacings and should not be drastically variable.
 -  **Use straightforward and easily understandable slider shapes.** New players may not understand how to follow more complex shapes.
 -  **Spinner recovery time should be at least 2 beats.** This is to ensure adequate time to click a hit object following a spinner.
 -  **Avoid spinners less than 3 beats.** Players need time to recognize that they have to begin spinning.
+-  **Avoid fully covering slider reverse arrows with hitobjects within 3 beats of the reverse arrow being reached.** This gives players enough time to recognize the reverse arrow.
```
Comment (cl8n) [inline wiki/Ranking_Criteria/osu!/en.md]:
```suggestion
-  **Avoid fully covering slider reverse arrows with hit objects within 3 beats of the reverse arrow being reached.** This gives players enough time to recognize the reverse arrow.
```

File: wiki/Ranking_Criteria/osu!/en.md [hunk 8]
```diff
@@ -57,27 +54,21 @@ Overall rules and guidelines apply to every kind of osu! difficulty. Rhythm rela
...
 -   **Spinners must be long enough for Auto to achieve 1000 bonus score.** Short spinners are unreasonably difficult to complete.
 -   **Each beatmap must use at least two different custom combo colours unless the default skin is forced.** The combo colours must not blend with the beatmap's background/storyboard/video in any case. This is so hit objects are always visible to the player and custom skin's combo colours do not blend with the background accidentally.
--   **Every slider must have a clear and visible path to follow from start to end.** Sliders which overlap themselves in a way that makes any of their individual sections unreadable or ambiguous cannot be used, such as burai sliders and hold sliders without straightforward slider borders.
+-   **Every slider must have a clear and visible path to follow from start to end.** Sliders that overlap themselves without straightforward slider borders and sliders whose individual sections are unreadable cannot be used. A slider's end position must be clear under the assumption that a player has fully transparent `sliderendcircle.png`/`sliderendcircleoverlay.png` skin elements.
```
Comment (dorothy3242) [inline wiki/Ranking_Criteria/osu!/en.md]: I disagree with specifically mentioning the skin elements here, as many may still use skins that use the transparent hitcircle and numbers as hitcircles to get the transparent sliderend style, as this is how it was done for years before seperate-end support was finally added. Just a particularity thing.
`A slider's end position must be clear under the assumption that a player has a skin which makes slider end circles invisible.`

File: wiki/Ranking_Criteria/en.md (hunk 1 above)
Comment (pishifat) [reply wiki/Ranking_Criteria/en.md]: adjusted

File: wiki/Ranking_Criteria/en.md (hunk 2 above)
Comment (pishifat) [reply wiki/Ranking_Criteria/en.md]: clayton makes the right decisions

File: wiki/Ranking_Criteria/osu!/en.md (hunk 8 above)
Comment (pishifat) [reply wiki/Ranking_Criteria/osu!/en.md]: yes

File: wiki/Ranking_Criteria/osu!/en.md (hunk 5 above)
Comment (pishifat) [reply wiki/Ranking_Criteria/osu!/en.md]: yes

File: wiki/Ranking_Criteria/osu!/en.md (hunk 3 above)
Comment (pishifat) [reply wiki/Ranking_Criteria/osu!/en.md]: yes? ill ask other mode players first tho, dunno what silly stuff they consider fine

File: wiki/Ranking_Criteria/osu!/en.md (hunk 4 above)
Comment (pishifat) [reply wiki/Ranking_Criteria/osu!/en.md]: burn

File: wiki/Ranking_Criteria/osu!/en.md (hunk 7 above)
Comment (pishifat) [reply wiki/Ranking_Criteria/osu!/en.md]: yes

File: wiki/Ranking_Criteria/osu!/en.md (hunk 6 above)
Comment (pishifat) [reply wiki/Ranking_Criteria/osu!/en.md]: yes

**Comment by pishifat** [comment]
best to wait on this until after https://github.com/ppy/osu-wiki/pull/2160 is merged

**Pull request merged**